.github/workflows/ci-lambda.yml

```diff
@@ -0,0 +1,161 @@
+      #       CHANGEDPUSHFILES=$(git diff --name-only ${{ github.sha }}^ ${{ github.sha }})
+      #       echo "Changed files: $CHANGEDPUSHFILES"
+      #       echo 'CHANGEDPUSHFILES='"${CHANGEDPUSHFILES}" >> $GITHUB_OUTPUT
+
+      - name: Determine changed files for pull request event
+        id: list-pr-changes
+        if: github.event_name == 'pull_request'
+        run: |
+            echo "Fetching list of changed files for pull request event..."
+            PR_NUMBER=${{ github.event.pull_request.number }}
+            PR_FILES=$(gh pr diff $PR_NUMBER --name-only)
+            echo "${PR_FILES}" > ${{ runner.temp }}/PRLIST.txt
+            echo "CHANGEDPRLIST=${{ runner.temp }}/PRLIST.txt" >> $GITHUB_OUTPUT
+
+      - name: Determine changed files for merge_group event
+        id: list-merge-changes
+        if: github.event_name == 'merge_group'
+        run: |
+            # Get target branch (base branch) name
+            BASE_BRANCH=$(jq -r '.pull_request.base.ref' < "$GITHUB_EVENT_PATH")
+            # Fetch base branch
+            git fetch origin $BASE_BRANCH
+            # Get merge base commit
+            BASE_COMMIT=$(git merge-base HEAD origin/$BASE_BRANCH)
+            echo "BASE_COMMIT=$BASE_COMMIT" >> $GITHUB_ENV
+            MERGE_FILES=$(git diff --name-only ${{ env.BASE_COMMIT }} HEAD)
+            echo "${MERGE_FILES}" > ${{ runner.temp }}/MRGLIST.txt
+            echo "CHANGEDMERGELIST=${{ runner.temp }}/MRGLIST.txt" >> $GITHUB_OUTPUT
+
+      - name: Delineate Modified Folders
+        id: validate-testworthy-changes
+        run: |
+            CHANGED_FILES=${{ runner.temp }}/CFLIST.txt
+            UNIQ_FOLDERS=${{ runner.temp }}/UNIQFLDR.txt
+            touch ${CHANGED_FILES}
+            if [ -f "${{ steps.list-merge-changes.outputs.CHANGEDMERGELIST }}" ]
+            then
+              echo "MERGE Change List"; cat ${{ steps.list-merge-changes.outputs.CHANGEDMERGELIST }}
+              cat ${{ steps.list-merge-changes.outputs.CHANGEDMERGELIST }} >> ${CHANGED_FILES}
+            fi
+            if [ -f "${{ steps.list-pr-changes.outputs.CHANGEDPRLIST }}" ]
+            then
+              echo "PR Change List"; cat ${{ steps.list-pr-changes.outputs.CHANGEDPRLIST }}
+              cat ${{ steps.list-pr-changes.outputs.CHANGEDPRLIST }} >> ${CHANGED_FILES}
+            fi
+            echo 'Changed files:'; cat ${CHANGED_FILES}
+            # the following sequence reduces CHANGED_FILES to CHANGEDIRS
+            # thereby avoiding multiple test cycles for a single folder of muliple file changes
+            cat ${CHANGED_FILES} | while read f
+            do
+              if [ -f $f ]
+              then
+                dirname $f
+              fi
+            done | sort | uniq >> ${UNIQ_FOLDERS}
+            UNIQ64=$(cat ${UNIQ_FOLDERS}|base64)
+            echo "FOLDERLIST=${UNIQ64}" >> $GITHUB_OUTPUT
+            echo "FOLDERLINES=$(cat ${UNIQ_FOLDERS}|wc -l)"
```

Comment:
Recommend taking a look at the [Changed Files](https://github.com/marketplace/actions/changed-files) Action on the GHA marketplace and replacing all the steps here with it. If you specify `dir_names: "true"` as a `with` parameter, it will return a `separator` (space by default, but a non-whitespace delimiter is recommended to avoid issues with spaces in paths) delimited list of _unique_ directory names in its `all_changed_files` output (full path `steps.changed-dirs.outputs.all_changed_files`), i.e:

```yaml
- name: Get changed files
  id: changed-dirs
  uses: tj-actions/changed-files@v44
  with:
    dir_names: "true"
    separator: ","
```

Should vastly simplify the `python-check` since there is no longer a need to compute the list of modified directories manually.

Reply:
reworked